docs/STRATEGY.md

```diff
@@ -0,0 +1,166 @@
+       example, if you are querying a database, it might be more appropriate to let the database server do the count for you,
+       rather than bringing the whole collection into memory and then counting using Ruby code.<br/>
+       **Important:** Just for your information and in order for you to be cautious on the implementation, if one calls `#count`,
+       on an `Enumerator` instance, this loads the whole collection into memory before returning the actual result.
+   2. `criteria_result = migration_strategy.criteria` needs to be implemented so that `criteria_result` responds
+      to `#each`.
+   2. `each_result` = `criteria_result.each` needs to be implemented so that `each_result` responds to `#next`
+   3. `next_result = each_result#next` is the next item that will be processed. And multiple calls to
+      `each_result#next` should bring the next item to be processed, or raise `StopIteration` exception if
+      no other item left in the collection.
+
+   The above may seem a little bit complicated, but it is actually easier than what it looks initially. This is because
+   the collections in Ruby expose, both a `#count` method and an `#each` method returning an `Enumerator` which supports the `#next`.
+   So, for example, if the `#criteria` method returns an `Array`, nothing more is required, because `Array#each`
+   returns an `Enumerator` and the `Enumerator#next` returns the next item lazy loaded.
+
+   See later on for two examples migration strategies. One that is using the MongoID interface to access a Mongo DB.
+   And another one using the Mongo Ruby Driver for the same database.
+
+1. Needs to respond to `#process(item)`. It should implement a `#process(item)` method that
+   will be called by the framework. The method needs to return `true` if the item is processed successfully,
+   or `false` if it is not processed successfully. This is taken into account by the framework to keep
+   track of the items that have been processed successfully. Also the framework makes sure to log the
+   correct lines in the log file accordingly.
+
+1. Optionally, responds to `#interrupt_callback`. This method will not be normally called. It will be called by
+   the framework if the user running the migration script interrupts it by hitting the key combination Ctrl+C or by sending
+   the signal `SIGTERM (15)` or `SIGINT (2)` (e.g. with `kill process_id` the `SIGTERM` is sent). Hence, the migration strategy
+   is responsible to carry out clean up actions for the premature termination of the migration script. It needs to make sure
+   that the script finishes gracefully.
+
+## Example Migration Strategies
+
+You can read here two example migration strategies for MongoDB.
+
+### Migration Strategy Using Mongo Ruby Driver
+
+``` ruby
+class MigrationStrategy
+
+  def process(policy)
+    sleep 1
+    true
+  end
+
+  def criteria
+    query
+  end
```

Comment:
can you swap the `#criteria` and `#process` around in the examples as well please?